examples/TestEm3/TestEm3.cu

```diff
@@ -193,10 +195,17 @@ void TestEm3(const vecgeom::cxx::VPlacedVolume *world, int numParticles, double
   COPCORE_CUDA_CHECK(cudaMemcpy(MCIndex_dev, MCIndex_host, sizeof(int) * numVolumes, cudaMemcpyHostToDevice));
   COPCORE_CUDA_CHECK(cudaMemcpyToSymbol(MCIndex, &MCIndex_dev, sizeof(int *)));
 
+  cudaDeviceProp deviceProp;
+  cudaGetDeviceProperties(&deviceProp, 0);
+
   // Capacity of the different containers aka the maximum number of particles.
-  constexpr int Capacity = 1024 * 1024;
+  // Use 2/7 of GPU memory for each of e+/e-/gammas, leaving 1/7 for the rest.
+  int Capacity = deviceProp.totalGlobalMem / sizeof(Track) * (2.0 / 7.0);
```

Comment:
So I realized that the choice of `Capacity` has a gigantic impact on the runtime, because it directly determines the number of batches that will be iterated on later. We should definitely separate this change from the rest, so we can benchmark the impact of the kernel separation alone.